client/me/form-base/with-form-base.jsx

```diff
@@ -8,96 +8,54 @@ import { connect } from 'react-redux';
 /**
  * Internal dependencies
  */
-import user from 'calypso/lib/user';
-import userSettings from 'calypso/lib/user-settings';
 import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import QueryUserSettings from 'calypso/components/data/query-user-settings';
+import getUserSettings from 'calypso/state/selectors/get-user-settings';
+import getUnsavedUserSettings from 'calypso/state/selectors/get-unsaved-user-settings';
+import hasUnsavedUserSettings from 'calypso/state/selectors/has-unsaved-user-settings';
+import {
+	setUserSetting,
+	clearUnsavedUserSettings,
+	saveUserSettings,
+} from 'calypso/state/user-settings/actions';
+import { isUpdatingUserSettings } from 'calypso/state/user-settings/selectors';
 
 const withFormBase = ( WrappedComponent ) => {
```

Comment:
Just a note that it may make sense to get rid of `withFormBase` eventually. I'm not sure the abstraction it provides is really worth it (which essentially means hiding functionality from the underlying component). I'll evaluate this claim after we got rid of the former `formBase` mixin.

Reply:
Good call - I'd love for us to get rid of the `formBase` and `withFormBase` when we're done with all these 👍  I also don't see any value in using them. And it gets even less generic and more specific (thus, less useful) as we added these user settings redux bits.